Tensile/KernelWriter.py

```diff
@@ -945,34 +959,42 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
         # can't do shadow initC with multiple summation since this resets the ValuC counters
         # on each unroll iteration.
         self.doShadowInit = 1 # 1 is just store setup
+
+    # in HPA mode, can do the alpha, beta conversion f16->f32 only once
+    kl.append( self.checkAlphaBetaForHPA(kernel))
+
     if self.prefetchAcrossPersistent:
       # first prefetch is outside persistent loop, subsequent prefetch will
       # be integrated into no-load-loop
-      kl += self.setupNewTile(kernel, tensorParametersA, tensorParametersB, False)
+      kl += self.setupNewTile(kernel, tensorParametersA, tensorParametersB, isPap=False, isOptNLL=False)
       kl.append(self.openPersistentLoop(kernel))
     else:
       # prefetch is inside persistent loop
       kl.append(self.openPersistentLoop(kernel))
-      kl += self.setupNewTile(kernel, tensorParametersA, tensorParametersB, False)
+      kl += self.setupNewTile(kernel, tensorParametersA, tensorParametersB, isPap=False, isOptNLL=False)
 
     pack = [ Code.Module() for i in range (self.numItersPLR+1) ]
 
     if kernel["PrefetchGlobalRead"]:
+      pkWithStoreRemap = kernel["PersistentKernel"] and kernel["StoreRemapVectorWidth"]
       if self.doShadowInit:
         kl.append(self.openShadowInit(kernel))
         kl.append(self.globalWriteWorkGroupInit(kernel))
         if self.doShadowInit == 2:
           kl.append(self.initC(kernel)) # initC while waiting for global reads
+        if self.enable["Wait"]:
+          kl.append(self.wait(kernel, tensorParametersA, tensorParametersB, 0, -1, -1, "8wait for global read"))
+          # These cases loop back and run the prefetch loop again
+          # we need an extra barrier to ensure that the ds_reads from previous iteration
+          # have finished before we generate the prefetch for the next summation index.
+          if pkWithStoreRemap or self.actualSummationLoops>1:
+            kl.append( self.indent + self.syncStr + "// for PersistentKernel with SRVW " + self.endLine )
         kl.append(self.closeShadowInit(kernel))
-
-      if self.enable["Wait"]:
-        kl.append(self.wait(kernel, tensorParametersA, tensorParametersB, 0, -1, -1, "8wait for global read"))
-
-        # These cases loop back and run the prefetch loop again
-        # we need an extra barrier to ensure that the ds_reads from previous iteration
-        # have finished before we generate the prefetch for the next summation index.
-        if kernel["PrefetchAcrossPersistent"] or self.actualSummationLoops>1:
-          kl.append(self.syncThreads(kernel))
+      else: # no doShadowInit
+        if self.enable["Wait"]:
```

Comment:
change the behavior of non-pk and would shift wait global load earlier

Reply:
Yes, I found this is redundant modification. Fixed.